spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java

```diff
@@ -195,4 +197,12 @@ public void testTemplateDisambiguation() throws Exception {
 		pf.createProducer().close();
 	}
 
+	@Test
+	public void flushWithoutSend() throws Exception {
+		DefaultKafkaProducerFactory<String, String> pf = new DefaultKafkaProducerFactory<String, String>(Collections.emptyMap());
+		KafkaTemplate<String, String> template = new KafkaTemplate<String, String>(pf);
+		assertThatExceptionOfType(IllegalStateException.class)
+				.isThrownBy(() -> template.flush())
+				.withMessageContaining("'producer' must not be null for flushing.");
+	}
```

Comment:
Wow! That's much better than my variant :smile: 

Accepting it even I merged already.